File: backend/backend/src/main/java/com/github/beatrizgomees/contact/project/domain/ContactModel.java
package com.github.beatrizgomees.contact.project.domain;


import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Objects;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "contacts")
@Data
public class ContactModel {
    @Id
    String objectId;
    String name;
    String phone;
    String email;
    String photo;
    boolean favorite = false;
    boolean archive = false;


    public String getObjectId() {
        return objectId;
    }

    public void setObjectId(String objectId) {
        this.objectId = objectId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public boolean isFavorite() {
        return favorite;
    }

    public void setFavorite(boolean favorite) {
        this.favorite = favorite;
    }

    public String getPhoto() {
        return photo;
    }

    public void setPhoto(String photo) {
        this.photo = photo;
    }

    public boolean isArchive() {
        return archive;
    }

    public void setArchive(boolean archive) {
        this.archive = archive;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ContactModel that = (ContactModel) o;
        return favorite == that.favorite && archive == that.archive && Objects.equals(objectId, that.objectId) && Objects.equals(name, that.name) && Objects.equals(phone, that.phone) && Objects.equals(email, that.email) && Objects.equals(photo, that.photo);
    }

    @Override
    public int hashCode() {
        return Objects.hash(objectId, name, phone, email, photo, favorite, archive);
    }
}

File: frontend/contact_book_app/lib/features/contact/model/contact_model.dart
class ContactModel {
   String? objectId;
   String? name;
   String? phone;
   String? email;
   String? password;
   String? photo;
   bool? favorite = false;
   bool? archive = false;

 ContactModel({
  this.objectId,
  this.name,
  this.password,
  this.email,
  this.phone,
  this.photo,
  this.favorite = false,
  this.archive = false
    });
  
  
  
  factory ContactModel.fromMap(Map<String, dynamic> data) {
  return ContactModel(
    objectId: data['objectId'],
    name: data['name'],
    phone: data['phone'],
    photo: data['photo'],
    email: data['email'],
    password: data['password'],
    favorite: data['favorite'],
    archive: data['archive']
    
  );
}

  Map<String, dynamic> toJson() {
    return {
      'objectId': objectId,
      'name': name,
      'phone': phone,
      'email': email,
      'password': password,
      'photo': photo,
      'favorite': favorite,
      'archive': archive
    };
  }
}
